### PR TITLE
Added BookProgress tests

### PR DESCRIPTION
**Pull request by bonnieli0812**
Added BookProgressMAO and refactored BookProgressService to make it more testable
Amended some BookProgressService methods because they were shit
Reverted changes to misc.xml and vcs.xml from #103 (imo we should just keep the JDK/SDK the same, it's been working for us so far)
Made some temporary fixes to Book (the implementation of the genre field is probably a bit undercooked, was breaking some stuff - a new task should be made to address it)
Added BookProgress unit tests

### CODE REVIEW

File: src/main/java/com/bookclub/service/BookProgressService.java [hunk 1]
```diff
@@ -48,7 +53,7 @@ public boolean saveBookProgress(Book book, User user, int pageNumber) {
         if (bookProgress != null) {
             bookProgress.setPageNumber(pageNumber);
         }
-        return bookProgress != null ? bookProgressAO.updateBookProgress(bookProgress) : startBookProgress(book, user);
+        return bookProgress != null ? bookProgressAO.updateBookProgress(bookProgress) : bookProgressAO.addBookProgress(new BookProgress(book.getId(), user.getId(), pageNumber));
```
Comment (jmccook96) [inline src/main/java/com/bookclub/service/BookProgressService.java:56]: 😬 she's getting a lil long

File: src/test/java/com/example/BookProgressServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        bookProgressMAO.addBookProgress(new BookProgress(book.getId(), 2, 3));
+        bookProgressMAO.addBookProgress(new BookProgress(book.getId(), 3, 1));
+
+        List<BookProgress> expected = List.of(new BookProgress(book.getId(), user.getId(), 10),
+                                              new BookProgress(book.getId(), 2, 3),
+                                              new BookProgress(book.getId(), 3, 1));
+        List<BookProgress> actual = bookProgressService.getBookProgressListForBook(book);
+
+        assertEquals(expected.size(), actual.size());
+        assertEquals(1, actual.get(0).getBookId());
```
Comment (jmccook96) [inline src/test/java/com/example/BookProgressServiceTest.java:148]: Should we not be using the expected list, you're currently just using it to check list sizes?

**Review by jmccook96 — APPROVED** [review]

File: src/test/java/com/example/BookProgressServiceTest.java (hunk 2 above)
Comment (bonnieli0812) [reply src/test/java/com/example/BookProgressServiceTest.java:144]: I tried to do assertEquals(expected, actual) with the lists but it wouldn't cooperate even though they had the same items so I've done the comparison manually.

I have updated to remove the expected list (wasn't being used anyway)

File: src/main/java/com/bookclub/service/BookProgressService.java (hunk 1 above)
Comment (bonnieli0812) [reply src/main/java/com/bookclub/service/BookProgressService.java:56]: *how can i add more to make it longer*

**Pull request merged**